backend/routes/products.js

```diff
@@ -2,43 +2,42 @@ const express = require("express");
 const router = express.Router();
 const Product = require("../models/Product");
 
-// return list of products
+// return list of products (get request)
 router.get("/", async (req, res) => {
   const productsData = await Product.find({});
   res.send({ productsData });
 });
 //
 
-// post request for new products
+// add a new product to the DB (post request)
 router.post("/add-new", async (req, res) => {
-  const title = req.body.title;
-  const image = req.body.image;
-  const description = req.body.description;
-  const category = req.body.category;
-  const price = req.body.price;
+  const { title, image, description, category, price } = req.body;
 
   if (!title || !image || !description || !category || !price) {
     res
       .status(400)
       .send({ message: "Please provide all the necessary details" });
   }
 
-  const productExists = await Product.findOne({ title: title });
+  // const productExists = await Product.findOne({ title: title });
 
-  if (productExists) {
+  const productExists = await Product.find({ title }).limit(1).lean();
+
+  if (productExists.length > 0) {
     res
       .status(400)
       .send("Product with following title is already in the database.");
   } else {
     const product = new Product({ title, image, description, category, price });
 
-    product.save((err, doc) => {
-      if (err) {
-        res.status(500).send("Product creation failed. please try again later");
-      }
+    const a = await product.save();
+    console.log(a);
 
+    if (a) {
```

Comment:
same thing as before..

start with if(!a) return error
then u dont need to use else and code looks clean